bitcoin/src/blockdata/transaction.rs

```diff
@@ -789,6 +790,41 @@ pub fn effective_value(
     (value.to_signed() - fee.to_signed()).expect("cannot overflow")
 }
 
+/// Computes the absolute value of an output given an effective value.
+///
+/// # Parameters
+///
+/// * `fee_rate` - the fee rate of the transaction being created.
+/// * `input_weight_prediction` - the predicted input weight.
+/// * `effective_value` - The value an output minus the fees to spend it.
+///
+/// # Returns
+///
+/// * [`NumOpResult::Error`] if the effective_value is negative since an absolute
+///   value must be positive.
+/// * [`NumOpResult::Error`] if fee plus effective_value causes an overflow.
+/// * [`NumOpResult::Valid`] on successful calculation.
```

Comment:
```suggestion
/// Computes the full output value accounting for the cost of spending it.
///
/// The absolute value is the value of an output plus the amount required to spend it. That is, the
/// absolute value can be calculated as: effective_value + (fee_rate * weight).
///
/// This function essentially reverses the effective value calculation by adding back the estimated
/// spending fee based on the fee rate and input weight.
///
/// # Parameters
///
/// * `fee_rate` - the fee rate of the transaction being created.
/// * `input_weight_prediction` - the predicted input weight.
/// * `effective_value` - The effective value of the output (output value minus spending fee).
///
/// # Returns
///
/// * [`NumOpResult::Error`] if the effective value is negative or if overflow occurs.
/// * [`NumOpResult::Valid`] with the full output value on success.
```

It might help to clarify the description by explaining that `absolute_value` adds back the spending fee to the effective value, similar to how the `effective_value` doc explains its calculation.